### PR TITLE
add two more drive metrics when metrics is available

### PR DESCRIPTION
**Pull request by harshavardhana**
## Contribution License
All contributions in this pull request are licensed to the project maintainers 
under the terms of the [Apache 2 license] (https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description
add two more drive metrics when metrics are available

## Motivation and Context
the new change also renames existing metrics as well
from disk -> drive

## How to test this PR?
Exposes drive-level error counters, and API-level latencies.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

**Review by jiuker — APPROVED** [review]
code LGTM

File: cmd/metrics-v2.go [hunk 1]
```diff
@@ -389,17 +387,37 @@ func getNodeDriveAPILatencyMD() MetricDescription {
...
 	}
 }
 
+func getNodeDriveTimeoutErrorsMD() MetricDescription {
+	return MetricDescription{
+		Namespace: nodeMetricNamespace,
+		Subsystem: driveSubsystem,
+		Name:      "errors_timeout",
+		Help:      "Total number of timeout errors since server start",
+		Type:      gaugeMetric,
```
Comment (vadmeste) [inline cmd/metrics-v2.go]: This looks more like a counterMetric

File: cmd/metrics-v2.go [hunk 2]
```diff
@@ -389,17 +387,37 @@ func getNodeDriveAPILatencyMD() MetricDescription {
...
+	}
+}
+
+func getNodeDriveAvailablityErrorsMD() MetricDescription {
+	return MetricDescription{
+		Namespace: nodeMetricNamespace,
+		Subsystem: driveSubsystem,
+		Name:      "errors_availability",
+		Help:      "Total number of I/O errors, permission denied and timeouts since server start",
+		Type:      gaugeMetric,
```
Comment (vadmeste) [inline cmd/metrics-v2.go]: This looks more like a counterMetric

File: cmd/metrics-v2.go [hunk 3]
```diff
@@ -389,17 +387,37 @@ func getNodeDriveAPILatencyMD() MetricDescription {
...
+		Help:      "Total number of timeout errors since server start",
+		Type:      gaugeMetric,
+	}
+}
+
+func getNodeDriveAvailablityErrorsMD() MetricDescription {
+	return MetricDescription{
+		Namespace: nodeMetricNamespace,
+		Subsystem: driveSubsystem,
+		Name:      "errors_availability",
```
Comment (vadmeste) [inline cmd/metrics-v2.go:411]: maybe 'live_errors' name is better?

**Review by vadmeste — COMMENTED** [review]
Few comments

File: cmd/metrics-v2.go (hunk 3 above)
Comment (harshavardhana) [reply cmd/metrics-v2.go:411]: no these are errors for data availability @vadmeste not `live`

**Review by vadmeste — APPROVED** [review]
LGTM

**Review by poornas — APPROVED** [review]

**Pull request merged**